rita/src/rita_common/rita_loop/mod.rs

```diff
@@ -100,32 +108,48 @@ impl Handler<Tick> for RitaLoop {
                 .then(move |res, act, _ctx| {
                     let res = res.unwrap().unwrap();
 
-                    info!("got neighbors: {:?}", res);
+                    info!("TunnelManager got tunnels: {:?}", res);
 
                     let neigh = Instant::now();
 
-                    for &(ref their_id, _, ref ip) in &res {
-                        TunnelManager::from_registry()
-                            .do_send(OpenTunnel(their_id.clone(), ip.clone()));
-                    }
-
                     let res = res
                         .iter()
-                        .map(|input| (input.0.clone(), input.1.clone()))
+                        .map(|res| (res.0.clone(), res.1.clone()))
                         .collect();
 
                     TrafficWatcher::from_registry()
                         .send(Watch(res))
                         .into_actor(act)
                         .then(move |_res, _act, _ctx| {
                             info!("loop completed in {:?}", start.elapsed());
-                            info!("traffic watcher completed in {:?}", neigh.elapsed());
+                            info!("TrafficWatcher completed in {:?}", neigh.elapsed());
                             DebtKeeper::from_registry().do_send(SendUpdate {});
                             PaymentController::from_registry().do_send(PaymentControllerUpdate {});
                             actix::fut::ok(())
                         })
                 }),
         );
+
+        trace!("Starting PeerListener tick");
+        Arbiter::spawn(
+            PeerListener::from_registry()
+                .send(Tick {})
+                .then(|res| {
+                    trace!("PeerListener said {:?}", res);
+                    res
+                }).then(|_| Ok(())),
+        );
+
+        trace!("Getting Peers from PeerListener to pass to TunnelManager");
+        Arbiter::spawn(
+            PeerListener::from_registry()
+                .send(GetPeers {})
+                .and_then(|peers| {
+                    trace!("Got peers from PeerListener, passing to TunnelManager");
+                    TunnelManager::from_registry().send(PeersToContact(peers.unwrap())) // GetPeers never fails so unwrap is safe
```

Comment:
If it never fails then why does it need a `Result`? (unless it's forced by actix?)

Reply:
forced by actix